pyproject.toml

```diff
@@ -1,10 +1,25 @@
 [build-system]
 # Setuptools version should match setup.py; wheel because pip will insert it noisily
-requires = ["setuptools >= 38.3.0", "wheel"]
+requires = ["setuptools >= 42.0.0", "versioningit ~= 0.1.0", "wheel"]
```

Comment:
FWIW, didn't try yet, but looks really neat in that it indeed seems to avoid needing all the copied code etc as versioneer does.  That though then makes projects rely on the 'stability' of versiongit but indeed `~=` (or even `==` for paranoids) pinning should provide remedy